pkg/client/inspect_extension.go

```diff
@@ -0,0 +1,75 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/buildpacks/pack/pkg/buildpack"
+	"github.com/buildpacks/pack/pkg/dist"
+	"github.com/buildpacks/pack/pkg/image"
+)
+
+type ExtensionInfo struct {
+	Extension dist.ModuleInfo
+	Location  buildpack.LocatorType
+}
+
+type InspectExtensionOptions struct {
+	ExtensionName string
+	Daemon        bool
+}
+
+func (c *Client) InspectExtension(opts InspectExtensionOptions) (*ExtensionInfo, error) {
+	locatorType, err := buildpack.GetLocatorType(opts.ExtensionName, "", []dist.ModuleInfo{})
+	if err != nil {
+		return nil, err
+	}
+	var layerMd dist.ModuleLayers
+
+	layerMd, err = metadataOfExtensionFromImage(c, opts.ExtensionName, opts.Daemon)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &ExtensionInfo{
+		Extension: extractExtension(layerMd),
+		Location:  locatorType,
+	}, nil
+}
+
+func metadataOfExtensionFromImage(client *Client, name string, daemon bool) (layerMd dist.ModuleLayers, err error) {
+	imageName := buildpack.ParsePackageLocator(name)
+	img, err := client.imageFetcher.Fetch(context.Background(), imageName, image.FetchOptions{Daemon: daemon, PullPolicy: image.PullNever})
+	if err != nil {
+		return dist.ModuleLayers{}, err
+	}
+
+	if _, err := dist.GetLabel(img, dist.ExtensionLayersLabel, &layerMd); err != nil {
+		return dist.ModuleLayers{}, fmt.Errorf("unable to get image label %s: %q", dist.BuildpackLayersLabel, err)
+	}
+
+	return layerMd, nil
+}
+
+func extractExtension(layerMd dist.ModuleLayers) dist.ModuleInfo {
+	result := dist.ModuleInfo{}
+	extensionSet := map[*dist.ModuleInfo]bool{}
```

Comment:
I think this function could be made simpler by removing this map. I believe it's used in `extractBuildpacks` so that we can do `sort.Slice` on the result, but since we're always just returning one element we don't need to sort. We could maybe do a check on the length of the `layerMd` provided and if there is more than one, throw an error.